### PR TITLE
feat: add support for global config

### PR DESCRIPTION
**Pull request by nchamo**
**tl;dr;**
* We are adding a new source of configuration. This new source is an endpoint that is polled every 15 min
* We are adding a new endpoint `/config` to read server's the current configuration
* We are adding listeners to the `configService`, so other services can react to the change in config. By doing this, we are avoiding the need for whole server restarts

### What are we doing here?
We are now giving a lot more love to the `configService`. There will be 4 different sources for configuration, with the following precedence:
```
Environment config >> storage config >> global config >> defaults
```

The main change is that now, there is a global configuration. This configuration can be controlled by the foundation, but it can be easily overridden by catalyst owners. This approach will make it easier to test different configuration 
without having to connect to the server and restart it.

We are also created the concept of `LighthouseConfig`, so we avoid strings that can lead to problems with typos

_Note: this change was tested in .zone, by changing the global config and making sure that the server updates its config correctly._

### CODE REVIEW

**Review by pablitar — APPROVED** [review]
LGTM! Great work!
